README.md

```diff
@@ -110,89 +115,140 @@ trainer.train()
 
 ### `RewardTrainer`
 
-This is a basic example of how to use the `RewardTrainer` from the library. The `RewardTrainer` is a wrapper around the `transformers` Trainer to easily fine-tune reward models or adapters on a custom preference dataset.
+Here is a basic example on how to use the `RewardTrainer`:
 
 ```python
 # imports
+from trl import RewardConfig, RewardTrainer
+from trl.extras.dataset_formatting import conversations_formatting_function
+from datasets import load_dataset
 from transformers import AutoModelForSequenceClassification, AutoTokenizer
-from trl import RewardTrainer
 
-# load model and dataset - dataset needs to be in a specific format
-model = AutoModelForSequenceClassification.from_pretrained("gpt2", num_labels=1)
-tokenizer = AutoTokenizer.from_pretrained("gpt2")
-
-...
+# load model and tokenizer
+tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-0.5B-Instruct")
+model = AutoModelForSequenceClassification.from_pretrained(
+    "Qwen/Qwen2.5-0.5B-Instruct", num_labels=1
+)
+model.config.pad_token_id = tokenizer.pad_token_id
+
+# load dataset and preprocess
+dataset = load_dataset("trl-lib/Capybara-Preferences", split="train")
+
+def preprocess_function(examples):
```

Comment:
Note to self: we should align the reward trainer to be the same as the other ones where this preprocessing step is internalised